merlin/models/tf/transforms/sequence.py

```diff
@@ -348,3 +349,341 @@ def call(
+        # Ignoring masked items in the padding positions
+        output = tf.logical_and(output, padding_mask)
+        return output
+
+    @staticmethod
+    def _get_one_masked(row_lengths: tf.Tensor):
+        """Generates a dense mask boolean tensor where for each tensor (row)
+        there is exactly one True value (selected target)
+        """
+        max_seq_length = tf.cast(tf.reduce_max(row_lengths), tf.int32)
+        random_targets_indices = tf.cast(
+            tf.math.floor(
+                (
+                    tf.random.uniform(
+                        shape=tf.shape(row_lengths), minval=0.0, maxval=1.0, dtype=tf.float32
+                    )
+                    * tf.cast(row_lengths, tf.float32)
+                )
+            ),
+            tf.int32,
+        )
+
+        one_target_mask = tf.cast(tf.one_hot(random_targets_indices, max_seq_length), tf.bool)
+        return one_target_mask
+
+    def get_config(self):
+        config = super().get_config()
+        config["masking_prob"] = self.masking_prob
+        return config
+
+    @classmethod
+    def from_config(cls, config):
+        config = tf_utils.maybe_deserialize_keras_objects(config, ["pre", "post", "aggregation"])
+        schema = schema_utils.tensorflow_metadata_json_to_schema(config.pop("schema"))
+        target = config.pop("target")
+        masking_prob = config.pop("masking_prob")
+        return cls(schema, target, masking_prob, **config)
+
+
+@tf.keras.utils.register_keras_serializable(package="merlin.models")
+class MaskSequenceEmbeddings(Block):
+    """Takes a 3D input tensor (batch size x seq. length x embedding dim) and replaces
+    by a dummy trainable single embedding at the positions to be masked.
+    This block looks for the Keras mask (`._keras_mask`) in the following order:
+      1. Checks if the input tensor has a mask
+      2. Checks if there is a single target and if it has a mask
+      3. If there are multiple targets (dict) returns the mask of the target
+      that matches the first 2 dims of the input
+    This is useful to be used when PredictMasked() transformation is used in
+    the Loader, which randomly selects some targets to be predicted and uses
+    Keras Masking to cascade the `_keras_mask`. By replacing input embeddings
+    at masked positions we avoid target leakage when training models with
+    Masked Language Modeling (BERT-like)
+    """
+
+    def build(self, input_shape):
+        if len(input_shape) != 3:
+            raise ValueError("The inputs must be a 3D tensor (batch_size, seq_length, vector_dim)")
+        self.hidden_size = input_shape[-1]
+        if self.hidden_size is None:
+            raise ValueError("The last dim of inputs cannot be None")
+        # Create a trainable embedding to replace masked interactions
+        initializer = tf.random_normal_initializer(mean=0.0, stddev=0.001)
+        self.masked_item_embedding = tf.Variable(
```

Comment:
`item` is a bit of an overloaded term in Recsys. Would something like `masked_timestep_embedding` be a better name?